src/users.ts

```diff
@@ -0,0 +1,874 @@
+     * @returns {Object} - Access and Refresh Token.
+     * @example
+     * const user = {
+     *   "identity": "c52d-3b0d-43b9-8c3e-275c087d875af"
+     * }
+     *
+     */
+
+    const options: RequestInit = {
+      method: 'POST',
+      headers: {
+        'Content-Type': this.contentType,
+        Authorization: `Bearer ${refreshToken}`
+      },
+      body: JSON.stringify(login)
+    }
+
+    try {
+      const response = await fetch(
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+        new URL(
+          `${this.usersEndpoint}/tokens/refresh`,
+          this.usersUrl
+        ).toString(),
+        options
+      )
+      if (!response.ok) {
+        const errorRes = await response.json()
+        throw this.userError.HandleError(errorRes.error, response.status)
+      }
+      const tokenData = await response.json()
+      return tokenData
+    } catch (error) {
+      throw error
+    }
+  }
+
+  public async Update (user: User, token: string): Promise<User> {
+    // Update a user
+    /**
+     * @method Update - Update a user. Updates a user's name and metadata.
+     * @param {Object} user - User object.
+     * @param {String} token - Access token.
+     * @returns {Object} - User object.
+     * @example
+     * const user = {
+     * "id": "c52d-3b0d-43b9-8c3e-275c087d875af",
+     * "name": "John Doe"
+     * }
+     *
+     */
+
+    const options: RequestInit = {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': this.contentType,
+        Authorization: `Bearer ${token}`
+      },
+      body: JSON.stringify(user)
+    }
+
+    try {
+      const response = await fetch(
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+        new URL(
+          `${this.usersEndpoint}/${user.id}`,
+          this.usersUrl
+        ).toString(),
+        options
+      )
+      if (!response.ok) {
+        const errorRes = await response.json()
+        throw this.userError.HandleError(errorRes.error, response.status)
+      }
+      const userData = await response.json()
+      return userData
+    } catch (error) {
+      throw error
+    }
+  }
+
+  public async UpdateUserIdentity (user: User, token: string): Promise<User> {
+    // Update a user identity
+    /**
+     * @method UpdateUserIdentity - Update a user identity for a currently logged in user.
+     * The user Identity is updated using authorization user_token
+     * @param {Object} user - User object.
+     * @param {String} token - Access token.
+     * @returns {Object} - User object.
+     * @example
+     * const user = {
+     * "id": "c52d-3b0d-43b9-8c3e-275c087d875af",
+     * "credentials": {
+     *  "identity": "fkatwigs@email.com"
+     * }
+     *
+     * }
+     */
+
+    const options: RequestInit = {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': this.contentType,
+        Authorization: `Bearer ${token}`
+      },
+      body: JSON.stringify(user)
```

Comment:
Magistrala only requires a string called identity, not the whole user. You can do sth like:
```suggestion
       body: JSON.stringify({ identity: user.credentials?.identity })
```